Gems/Terrain/Code/Source/TerrainRenderer/TerrainClipmapManager.h

```diff
@@ -39,16 +39,24 @@ namespace Terrain
         static constexpr uint32_t DetailClipmapStackSizeMax = 16;
         static constexpr uint32_t SharedClipmapStackSizeMax = AZStd::max(ClipmapConfiguration::MacroClipmapStackSizeMax, ClipmapConfiguration::DetailClipmapStackSizeMax);
 
+        enum ClipmapSize : uint32_t
+        {
+            ClipmapSize512 = 512u,
+            ClipmapSize1024 = 1024u,
+            ClipmapSize2048 = 2048u
+        };
+
         //! The size of the clipmap image in each layer.
-        uint32_t m_clipmapSize = 1024u;
+        ClipmapSize m_clipmapSize = ClipmapSize1024;
```

Comment:
It seems like this or some related parameter should get exposed as well. Right now, we can say "I want 2048 texels/m, and I want to draw for 256 m", but we don't have an exposed control that says how many meters out we're drawing that 2048 texels/m for. It seems like we should either expose the clipmap size or else some sort of max resolution draw distance. The latter seems friendlier to work with as a user, but possibly problematic because the max texture limit of 4k (or 8k?) would cause this to have complicated cross-dependencies between settings, so it's probably better/easier just to expose the clipmap size and let the user do the math on how far that max resolution draw distance will go.